merkle/compact/tree.go

```diff
@@ -78,12 +70,20 @@ func NewTreeWithState(hasher hashers.LogHasher, size int64, getNodesFn GetNodesF
 	if got, want := len(hashes), len(ids); got != want {
 		return nil, fmt.Errorf("got %d hashes, needed %d", got, want)
 	}
-	for i, id := range ids {
-		r.nodes[id.Level] = hashes[i]
+	// Note: Right border nodes of compact.Range are ordered top-to-bottom.
```

Comment:
Might be useful to indicate which is the root, given the inconsistency we've had with terms like height and depth.